### PR TITLE
feat(application-system): Events used as log markers for application history

### PR DESCRIPTION
**Pull request by obmagnusson**
# ...
 A small refactor of the history functionality
## What

Using events to mark logs. 

## Why

Previous desing did not account for some states going back and forth with different actions.  Resulting in messages that would not apply.

## Screenshots / Gifs

Attach Screenshots / Gifs to help reviewers understand the scope of the pull request

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Formatting passes locally with my changes
- [ ] I have rebased against main before asking for a review

### CODE REVIEW

**Review by Toti91 — APPROVED** [review]
Sweet i like these changes

File: libs/application/api/history/src/lib/history.service.ts [hunk 1]
```diff
@@ -22,8 +20,9 @@ export class HistoryService {
   async saveStateTransition(
```
Comment (Thorrik58) [inline libs/application/api/history/src/lib/history.service.ts:20]: do these need to be documented?

File: libs/application/templates/data-protection-complaint/src/lib/DataProtectionComplaintTemplate.ts [hunk 2]
```diff
@@ -39,6 +39,14 @@ const DataProtectionComplaintTemplate: ApplicationTemplate<
             shouldBePruned: true,
             whenToPrune: 5 * 60000, //5 minutes
           },
+          actionCard: {
+            historyLogs: [
```
Comment (Thorrik58) [inline libs/application/templates/data-protection-complaint/src/lib/DataProtectionComplaintTemplate.ts:43]: feel like these properties within the object could be more descriptive.

File: libs/application/templates/reference-template/src/lib/ReferenceApplicationTemplate.ts [hunk 3]
```diff
@@ -142,7 +142,12 @@ const ReferenceApplicationTemplate: ApplicationTemplate<
 
           actionCard: {
             description: m.draftDescription,
-            onExitHistoryLog: 'Umsókn send inn',
+            historyLogs: [
+              {
+                event: DefaultEvents.SUBMIT,
+                log: 'Umsókn send inn',
```
Comment (Thorrik58) [inline libs/application/templates/reference-template/src/lib/ReferenceApplicationTemplate.ts:148]: Werent these supposed to be translation strings?

**Review by Thorrik58 — CHANGES_REQUESTED** [review]
questions 🗡️

File: apps/application-system/api/migrations/20230504150905-history-add-exit-event.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
+'use strict'
+
+module.exports = {
+  up: (queryInterface, Sequelize) => {
+    return queryInterface.sequelize.transaction((t) =>
+      Promise.all([
+        queryInterface.addColumn('state_history', 'exit_event', {
```
[the remark this answers is not in the document]
Comment (obmagnusson) [reply apps/application-system/api/migrations/20230504150905-history-add-exit-event.js:7]: I think tracking the exit event should be sufficient. As we are logging on events in states. I will add a previous state for future proofing.

File: libs/application/api/history/src/lib/history.service.ts (hunk 1 above)
Comment (obmagnusson) [reply libs/application/api/history/src/lib/history.service.ts:20]: Idk think they are pretty descriptive as is. Plus this shoulndt be used outside these use cases in the system.

**Pull request merged**